src/game/client/draw/w3dlaserdraw.cpp

```diff
@@ -13,3 +13,314 @@
  *            LICENSE
  */
 #include "w3dlaserdraw.h"
+#include "assetmgr.h"
+#include "colorspace.h"
+#include "drawable.h"
+#include "laserupdate.h"
+#include "segline.h"
+#include "terrainlogic.h"
+#include "texture.h"
+#include "w3ddisplay.h"
+#include "w3dscene.h"
+
+void Get_Color_Components_Real(int color, float *red, float *green, float *blue, float *alpha)
+{
+    *alpha = ((color & 0xFF000000) >> 24) / 255.0;
+    *red = ((color & 0xFF0000) >> 16) / 255.0;
+    *green = ((color & 0xFF00) >> 8) / 255.0;
+    *blue = color / 255.0;

```

Comment:
Division by `int` as opposed to `double` would suffice.

Reply:
In this case you absolutely need to divide by float since the return values are float. Fixed to do float instead of double though.